api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java

```diff
@@ -24,6 +26,8 @@
 public class DirectoryServiceImpl implements DirectoryService {
 
   static final String APPLICATION_NAME = "All of Us Researcher Workbench";
+  private static final String ALLOWED = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()";
+  private static SecureRandom rnd = new SecureRandom();
```

Comment:
I don't think we really need the cryptographically superior `SecureRandom` vs plain old `java.util.Random`. Either works for me.